### PR TITLE
Assessments tab: refactor answer indexes

### PR DESCRIPTION
**Pull request by caleybrock**
## User Impact
This is a pure refactor and doesn't affect any code currently visible to users. Must have internal `assessmentsTab` experiment on.

## Goal
The goal is to make the assessment tab code less fragile and easier to read (more internally consistent).

## Before
Responses API returned a string of letters `'A'` or `'B, C'` for student answers, but the questions API returned an ordered array where the index indicated what letter the answer corresponded to. This required calculations back and forth between indexes and letters on the client side. 

## After
Responses API now returns an array of indexes for student answers, which match the indexes we get in the questions API. Now the client always converts from internal indexes to "display answers" like `[0]` to `'A'` and `[3, 4]` to `'D, E'`. This makes the code easier to reason about and test.

This PR adds tests for helper methods added and updates controller and redux tests where data representation was changed.

### CODE REVIEW

File: apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js [hunk 1]
```diff
@@ -40,6 +40,8 @@ const MultiAnswerStatus = {
   INCORRECT: 'incorrect'
 };
 
+const ANSWER_LETTERS = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H'];
```
Comment (caleybrock) [inline apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js:43]: Do these need to be translated or are these always the values we use for assessments?

Comment (islemaster) [reply apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js:43]: :man_shrugging: Let's ask the curriculum team.

Comment (caleybrock) [reply apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js:43]: I'll leave this as a TODO and follow up with Poorva.

File: apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js [hunk 2]
```diff
@@ -471,44 +471,31 @@ export const getMultipleChoiceSectionSummary = (state) => {
...
-
-  const letterArr = ['A','B', 'C', 'D', 'E', 'F', 'G', 'H'];
   const correctLetters = [];
   for (let i = 0; i < answerArr.length; i++) {
     if (answerArr[i].correct) {
-      correctLetters.push(letterArr[i]);
+      correctLetters.push(ANSWER_LETTERS[i]);
     }
   }
   return correctLetters.join(', ');
```
Comment (islemaster) [inline apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js]: Could implement this function as:
```js
const correctIndexes = answerArr
  .map(({correct}, i) => (correct ? i : null))
  .filter(i => i !== null);
return return indexesToAnswerString(correctIndexes);
```
...but that might not be more readable.  Up to you.

File: apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js [hunk 3]
```diff
@@ -471,44 +471,31 @@ export const getMultipleChoiceSectionSummary = (state) => {
...
  * returns the corresponding letters to the options with the correct answers.
- *
- * TODO(caleybrock): Add letter options to response from the server so they are
- * consistent with the structure, but for now look up letter in this array.
- * If this code is left client side, it needs tests.
- *
  * Ex - [{correct: false}, {correct: true}] --> returns 'B'
  */
-const getCorrectAnswer = (answerArr) => {
+export const getCorrectAnswer = (answerArr) => {
```
Comment (islemaster) [inline apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js]: I actually prefer not to use arrow functions for top-level exports, where there's no `this` value to capture.  Mostly it's a style thing, but using a regular function declaration also binds the function name to the function object, which makes for more readable stack traces in some browsers (I think Chrome is clever enough that this doesn't matter anymore).

```js
export function getCorrectAnswer(answerArr) {
```

**Pull request merged**